src/backend/distributed/planner/extended_op_node_utils.c

```diff
@@ -214,7 +214,7 @@ HasNonPartitionColumnDistinctAgg(List *targetEntryList, Node *havingQual,
 		ListCell *varCell = NULL;
 		bool isPartitionColumn = false;
 
-		if (IsA(targetNode, Var))
+		if (IsA(targetNode, Var) || IsA(targetNode, GroupingFunc))
```

Comment:
How about specifically checking if the `targetNode` is an `Aggref`?

```suggestion
		if (!IsA(targetNode, Aggref))
```

Since we cast the pointer directly after this check it makes more sense to me to skip anything _but_ the `Aggref` instead of keep adding anything we find here that caused a crash.